django/sierra/base/models.py

```diff
@@ -1947,50 +1947,31 @@ def get_call_numbers(self):
                 call_number_tuples.extend(call_numbers)
         return call_number_tuples
 
-    def get_shelving_call_number(self):
+    def _cn_is_sudoc(self, cn_string, bib_cn_tuples):
         """
-        Returns one tuple that should represent the call number used
-        to shelve this item: (call_number, cn_type).
+        Takes a call number string and a list of bib call number
+        tuples. Returns True if the cn_string matches a sudoc number
+        in the list of bib_cn_tuples.
         """
-        call_number, cn_type = (None, None)
-        item_cn_tuples = self.get_call_numbers()
-        bib_cn_tuples = []
-        try:
-            bib_cn_tuples = (self.bibrecorditemrecordlink_set.all()[0]
-                                .bib_record.get_call_numbers())
-        except IndexError:
-            pass
-
-        # First: The first cn_tuple on the item record OR, if that
-        # doesn't exist, the first cn_tuple on the bib record
-        # is the shelving call number.
-
-        if len(item_cn_tuples) > 0:
-            (call_number, cn_type) = item_cn_tuples[0]
-        elif len(bib_cn_tuples) > 0:
-            (call_number, cn_type) = bib_cn_tuples[0]
-
-        # Second: Check the main shelving call number against any Gov
-        # Docs numbers that are on the bib. If the call number string
-        # matches a Gov Docs number, then the type should be "sudoc" no
-        # matter what.
-
-        if call_number and (call_number, 'sudoc') in bib_cn_tuples:
-            cn_type = 'sudoc'
-
-        # Third: Periodical call numbers. We might have a periodical
-        # shelved by title: if it has an ITYPE code of 5 (bound
-        # periodicals) and the call number is just two letters (like
-        # MT), or if the word "periodical" appears in the call number,
-        # or if there's ANY c-tagged field on the item with "SHELVED BY
-        # TITLE." In these cases, we get the Bib title and append it to
-        # the end of the call number.
+        is_sudoc = False
+        if cn_string and (cn_string, 'sudoc') in bib_cn_tuples:
+            is_sudoc = True
+        return is_sudoc
 
+    def _item_is_probably_shelved_by_title(self, cn_string):
+        """
+        Takes a call number string and tries to determine whether this
+        item is probably shelved by title. If the ITYPE is 5 (bound
+        periodicals) and the call number is just one or two letters
+        (like M or MT), or if the word "periodical" appears in the call
+        number string, or if there's ANY c-tagged field on the item
+        with "SHELVED BY TITLE." Returns a boolean value.
```

Comment:
missing the conclusion of those group of `or`s. so what happens if any of those are true? is it that the item is probably shelved?